cip/1.accepted/CIP2015-05-13-EXISTS.adoc

```diff
@@ -4,102 +4,188 @@
+A https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=RelationshipsPattern[<RelationshipsPattern>] _RP_ is effectively replaced by the expression
 
-For each matching subgraph evaluated with `EXISTS {}`, the result value must be `true` if the subquery finds at least one matching row.
-If no matches are found, `false` should be returned.
+`EXISTS { MATCH _RP_ RETURN 1 }`
 
-The subquery used in `EXISTS {}` has to follow the following rules:
+==== <SimpleExistentialSubquery>
 
- * It has to be a read-only query. Updating the graph as part of a predicate is not allowed.
- * It cannot end in a `RETURN` clause, nor may it contain `UNION`.
- * When `WITH` ends the subquery, it is able to change the cardinality of the subquery.
- In other words, since the `WITH` clause may include `WHERE`, `SKIP`, and/or `LIMIT`, the clause can with these sub-clauses turn a query that produces rows into one that does not.
- * It may use nested `EXISTS {}` predicates.
+A <SimpleExistentialSubquery> containing a <SimpleMatch> _SM_ is effectively replaced by the expression
 
-==== Simple subquery exists
+`EXISTS { MATCH _SM_ RETURN 1 }`
 
-When the subquery can be described with a single `MATCH-WHERE` clause, the `MATCH` keyword can be omitted, as in example 2A and B.
-The difference between this form and a simple pattern predicate, which is already available, is that this form allows for introducing new variables inside the `EXISTS {}` scope.
+==== <FullExistentialSubquery>
+
+A <FullExistentialSubquery> _FES_ is effectively evaluated as follows:
+
+* Let _OUTER_VARIABLES_ be the current working record for which the expression containing _FES_ is evaluated.
+* Let _NESTED_QUERY_ be the https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=RegularQuery[<RegularQuery>] immediately contained in _FES_.
+* Let _RESULT_TABLE_ be the table resulting from evaluating _NESTED_QUERY_ on a driving table comprising _OUTER_VARIABLES_.
+* Case:
+** If _RESULT_TABLE_ is an empty table (cardinality is zero), then the result of _FES_ is [underline]#*_False_*#.
+** Otherwise, the result of _FES_ is [underline]#*_True_*#.
+
+Note that all fields in _RESULT_TABLE_ are ignored and only the number of rows in _RESULT_TABLE_ is relevant for the result of _FES_.
 
 === Examples
 
-_Example 1A:_
+==== Property existence test
+
+_Example 1:_
 
 Return all nodes that have a property named `slogan`.
 [source, cypher]
 ----
 MATCH (actor)
-WHERE exists(actor.slogan)
+WHERE actor.slogan IS NOT NULL
 RETURN actor
 ----
 
+==== Pattern predicates in boolean expression context
+
 _Example 2A:_
 
+Find all actors who won an award.
+
+[source, cypher]
+----
+MATCH (actor:Actor) WHERE (actor)-[:WON]->(:Award)
+RETURN actor
+----
+
+_Example 2B:_
+
+Find all actors with their major accolade.
+
+[source, cypher]
+----
+MATCH (actor:Actor)
+RETURN actor,
+  CASE actor
```

Comment:
```suggestion
  CASE
```

When using the [Simple case form](https://neo4j.com/docs/cypher-manual/4.2/syntax/expressions/#syntax-simple-case) the expected type after `WHEN` is not exactly boolean.

Reply:
You are absolutely right. I meant the full case form but managed to get the syntax wrong.